continuous-testing/use-cases/markmap.md

```diff
@@ -1,29 +1,96 @@
+        * For Unit Testing, Integration, Dependency and Stub Mocking
+            * xUnit for build-time testing
             * Simulate dependencies
             * Stub out coupled services for planned results
             * Mock specific application behavior
             * Tools
                 * Java
+                    * JUnit
                     * Mockito
                     * EasyMock
                     * JMock
                 * Python
-                    * 
+                    * PyUnit
+                    * PyTest
+                    * Nose2
+                    * Mock
+                * C#
+                    * NUnit
+                    * Moq
+                    * NSubstitute
+                    * FakeItEasy
+                * C/C++
+                    * cUnit/CppUnit
+                    * CMock
+                    * cpp-stub
+                    * googletest
+    * For Code Analysis
+        * For Static Analysis
+            * Code formatting
+            * Application structure and layout
+            * Compliance with development standards
+            * Tools
+                * Java
+                    * PMD
+                    * Checkstyle
+                    * Spotbugs
+                * Python
+                    * PyLint
+                    * Pyflakes
+                    * Mypy
+                    * Prospector
+                    * PMD
                 * C#
+                    * StyleCop
+                    * Gendarme
+                    * Nitriq
+                    * PMD
                 * C/C++
-        * Code Quality Analysis
-            * Semmle
+                    * CppCheck
+                    * Clang Static Analyzer / tidy
+                    * CppDepend
+                    * Sourcetrail
+                    * PMD
+                    * FlawFinder
+        * For Dynamic Analysis
+            * Runtime performance
+            * Testing coverage
+            * Code complexity
+            * Tools
+                * Java
+                    * JProfiler
+                    * JMeter
+                    * Jacoco
+                    * Cobertura
+                    * JPF
+                * Python
+                    * Typo
+                    * Scalene
+                    * sys.settrace()/inspect/pdb
+                * C#
+                  * VS Debugger
+                  * Performance Profiler
+                  * Parasoft Ensure
+                  * ANTS Profiler
+                * C/C++
+                    * Gcov
+                    * Valgrind
+                    * LLVM/Clang
+        * For Code Quality (unified, both dynamic and static analyses)
+            * Unified analysis of static and dynamic code metrics
+            * Tools
+                * SonarQube
+                * Coverity
+        * For Security
+            * Verify and report vulnerabilities
+            * Check code security
+            * Tools
+                * Semmle
```

Comment:
CAE isn't supporting Semmle anymore. I'm not sure why. Microsoft Github did buy them out but it still appears to be offered through open source github. Regardless, JPL is migrating to the usage of scrub now (https://github.com/nasa/scrub).  Suggest adding scrub to the security tool list.

Reply:
Thanks. Will do.